### DISCUSSION TITLE
SentenceRecognizer - usage example is wrong

### DISCUSSION THREAD

**Discussion by Oxi84** [explosion/spaCy#11481] [Help: Coding & Implementations] [answered]
I do not really understand why would you write a 5 page guide to a software that does not actually say how to use this software. I mean code an examples. This   page - https://spacy.io/api/sentencerecognizer.  the following seemed like an incomplete example, but actually it does not work.

    import spacy
    nlp = spacy.load("en_core_web_sm")
    senter = nlp.add_pipe("senter")
    doc = nlp("This is a sentence.")

    ValueError: [E007] 'senter' already exists in pipeline. Existing names: ['tok2vec', 'tagger', 'parser', 'senter', 'attribute_ruler', 'lemmatizer', 'ner']

**Comment by svlandeg** [comment, marked as answer]
Hi, 

> I do not really understand why would you write a 5 page guide to a software that does not actually say how to use this software. I mean code an examples. This   page - https://spacy.io/api/sentencerecognizer. 

The spaCy documentation consists of two main parts: the [usage](https://spacy.io/usage) docs and the [API](https://spacy.io/api) docs. You want to start with the "usage" pages, in particular the page about [Language Processing Pipelines](https://spacy.io/usage/processing-pipelines). You'll find that throughout our documentation, we've inserted plenty of working code examples. The API docs are mainly for reference to more experienced users who want to know more about the specific ins and outs of the library.

> the following seemed like an incomplete example, but actually it does not work.
> 
> import spacy
> nlp = spacy.load("en_core_web_sm")
> senter = nlp.add_pipe("senter")
> doc = nlp("This is a sentence.")
> 

This code does not work because `en_core_web_sm` already has a `senter` component - as the error message clearly points out. If that code example would appear verbatim in the documentation, we'll be happy to adjust it. However I was unable to find it. Can you point me towards the exact location where you got this from?

Finally, I'd like to point out that we spend a lot of time and effort into writing this documentation. We're always open for constructive feedback on how to improve the docs for our users. However, your original post comes across as unnecessarily rude and disrepectful, so I'd like to ask you to please engage more constructively in future conversations. You can find spaCy's code of conduct [here](https://www.contributor-covenant.org/version/2/1/code_of_conduct/).

**Reply by Oxi84** [reply]
I know you took a lot of time into making software and documentation. I did not want to be rude of course, just pointing out you need examples of software usage, so people could use it. Sorry for that. Just want to make helpful suggestions.


This example is on this page: https://spacy.io/api/sentencerecognizer, just right of this line
[SentenceRecognizer.__call__ ](https://spacy.io/api/sentencerecognizer#call)method

    doc = nlp("This is a sentence.")
    senter = nlp.add_pipe("senter")
    # This usually happens under the hood
    processed = senter(doc)


Thanks

**Reply by svlandeg** [reply]
Right. This example will work when you use for instance
```
from spacy.lang.en import English
nlp = English()
```

**Reply by svlandeg** [reply]
By the way - you might also be interested in having a look at our free interactive course, which will get you going with the basic concepts more quickly: https://course.spacy.io/en
